packages/block-library/src/post-featured-image/edit.js

```diff
@@ -1,35 +1,40 @@
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import { useEntityProp } from '@wordpress/core-data';
 import { useSelect } from '@wordpress/data';
 import { ResponsiveWrapper } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
-function PostFeaturedImageDisplay() {
+function PostFeaturedImageDisplay( { context: { postId, postType } } ) {
 	const [ featuredImage ] = useEntityProp(
 		'postType',
-		'post',
-		'featured_media'
+		postType,
+		'featured_media',
+		postId
 	);
 	const media = useSelect(
 		( select ) =>
 			featuredImage && select( 'core' ).getMedia( featuredImage ),
 		[ featuredImage ]
 	);
-	return media ? (
+	if ( ! media ) {
```

Comment:
We need to show something to not confuse the users with no content at all. This is not displayed in front-end though. Maybe handle with a different message to let them know??

Any thoughts?

Reply:
We could probably draw inspiration from how we handle empty states elsewhere in Gutenberg by inviting the user to edit:

<img width="615" alt="Captura de ecrã 2020-09-25, às 17 46 13" src="https://user-images.githubusercontent.com/150562/94293990-076c5800-ff57-11ea-8441-c20cb9bb9f04.png">

But this is a general design problem, so /cc @mapk @jasmussen

Reply:
Thanks for the ping. This is what I see in this branch:

<img width="1988" alt="Screenshot 2020-09-28 at 08 26 41" src="https://user-images.githubusercontent.com/1204802/94397558-6521bf80-0164-11eb-928a-ff7762ae9685.png">

My instinct, when reading the context alone, was to suggest that we shouldn't show anything. Because just like an empty social links placeholder, if it's empty, there's nothing to show. 

However seeing it only in the site editor context and when editing templates, it definitely feels like a different heuristic is needed, one which does indeed show an empty state even if on some pages you won't add a featured image.

I would like us to eventually find a better design for the placeholders, one which better scales to small contexts, and also possibly looks a little bit like the block itself when finalized — right now it's just a sequence of white sheets with dark borders. And once we have that pattern it should be used for dropzones and placeholders alike. Here's a work in progress mockup for a dropzone in the Cover sidebar that I've abandoned because it's not very good:

<img width="646" alt="Screenshot 2020-09-28 at 08 29 40" src="https://user-images.githubusercontent.com/1204802/94398242-7fa86880-0165-11eb-9d5d-fa788efdc2dc.png">

<img width="406" alt="Screenshot 2020-09-28 at 08 28 34" src="https://user-images.githubusercontent.com/1204802/94398291-90f17500-0165-11eb-8712-ce43a0b39fc3.png">

However the point is the same — the placeholder should indicate the block that will land there, and can sometimes work as a dropzone. 

---

In the mean time, we should do what Miguel suggests, and just reuse the current placeholder material. But I'm not sure it's helpful to show that giant 200px tall sheet — because after all, there are posts where you won't add a featured image. So maybe go with a small chip for now:

<img width="1650" alt="Screenshot 2020-09-28 at 08 34 07" src="https://user-images.githubusercontent.com/1204802/94398471-e299ff80-0165-11eb-88c6-76ad2541ae5d.png">

Also, the text should be rephrased because in the site editor context, you _can't_ set a featured image, so suggesting you could is going to confuse. 

Also I wonder, is there a better name than "Post Featured Image"? Could it just be "Featured Image"?

Reply:
Thanks for reviewing @mcsf and @jasmussen!

>Also, the text should be rephrased because in the site editor context, you can't set a featured image, so suggesting you could is going to confuse.

Actually I think the first step forward is to show the previews at least and then give the ability to users to edit things here. This applies to content as well based on this comment here: https://github.com/WordPress/gutenberg/issues/25394#issuecomment-699083858.

The actual issue that this PR solves is that now on master the `featured image` when exist is not shown at all. I don't believe design and the `not set` placeholder should be a blocker for this PR and we can revisit with a follow up PR to handle the case of not having set a featured image. 

We can even show nothing for not having a featured image for now, as this is the current behavior. What do you think?

Reply:
Oh of course — that's valuable context — when the featured image is inside a query block, it will actually show the featured image of the _post queried_ 🤦  — yes absolutely agree, that needs to show the image directly, if set, and perhaps the chip above if not. 

> We can even show nothing for not having a featured image for now, as this is the current behavior. What do you think?

I would agree with your assessment that it's important to move forward with the baseline feature to get it working. But it highlights a really interesting template building question that's good to think about for future iteration, because even though showing nothing would seem correct in this case, the featured image "spot" seems like a unique situation. CC: @MichaelArestad @jameskoster @shaunandrews

Reply:
Hoo boy, this is a fun one. My current thoughts, fwiw:

If the template includes the featured image block then it should be visible in the site editor (in one form or another) _regardless_. 

When context is provided (e.g. I'm editing my singular template with the "Hello world" post loaded in to it) we should load the actual image if one is provided. This aligns with how other "token" blocks like the post title behave. Whether it should be possible to _edit_ the image in this scenario is less clear to me. It doesn't help that it is currently possible to edit the post _content_, but not other details like the _date_ or the _title_. It's a question of how much we want to blur the lines between template editing and content editing, and I feel we should be consistent in that behaviour. 

If I'm editing the same "Single" template with no context, the placeholder should be noninteractive and serve purely as a visual indication of where the block appears in the associated template.

Reply:
> When context is provided (e.g. I'm editing my singular template with the "Hello world" post loaded in to it) we should load the actual image if one is provided. This aligns with how other "token" blocks like the post title behave. Whether it should be possible to edit the image in this scenario is less clear to me. It doesn't help that it is currently possible to edit the post content, but not other details like the date or the title. It's a question of how much we want to blur the lines between template editing and content editing, and I feel we should be consistent in that behaviour.

Yeah that sums up the challenge. 

It looks like an image, but it isn't an image. However as soon as you select it, it should be clear that it's a featured image block. At least that feels sufficient to move forward to me.

What still feels vague to me is the no-content situation where you don't have any posts that add featured images, and you're only really building out the template. What shows up there? Given the insights from this discussion, perhaps it shouldn't be a dropzone. 

In that vein, will there ever be a situation where you are editing just a template, but with no content filled in? Seems like this would only ever happen on a brand new blog where actively delete pages and posts, no?

Reply:
>In that vein, will there ever be a situation where you are editing just a template, but with no content filled in?

I'm not sure for the current behavior, but that's something we'll need to be able to support and have a good design. 

>What still feels vague to me is the no-content situation where you don't have any posts that add featured images, and you're only really building out the template. What shows up there? Given the insights from this discussion, perhaps it shouldn't be a dropzone.

Based on the whole discussion I feel that the best choice for now is to have a simple non-editable placeholder like @jasmussen's screenshot above. The editable/non-editable `PostBlocks` is a different matter, that will handled in another issue and should definitely be consistent as @jameskoster has mentioned. For start I think the goal is to be non-editable, but we'll revisit again.

@jasmussen do you have a similar styled `chip` in mind I can see the styles, so as to be consistent?

Reply:
Yep seems good to move forward in a light weight way because we can always improve later.

The basic chip would be good to get.

https://github.com/WordPress/gutenberg/blob/e259fd03b7ba4852ef6b01be06ad4f5c4a100eec/packages/block-library/src/embed/editor.scss#L19

Lines 19-21 contain the css to make it work. I'd add some padding as well.

Reply:
Thanks! I added the chip.